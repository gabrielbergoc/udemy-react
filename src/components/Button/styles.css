.button {
  width: 100%;
  background: rgb(41, 41, 168);
  color: white;
  border: none;
  padding: 1rem 1.5rem;
  cursor: pointer;
}

.button:disabled {
  background: #888;
  cursor: not-allowed;
}
