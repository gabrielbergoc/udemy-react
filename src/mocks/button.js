export const buttonProps = {
  text: 'button text',
  title: 'button title',
  onClick: jest.fn(),
  disabled: false,
};
