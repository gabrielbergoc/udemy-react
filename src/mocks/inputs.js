export const searchInputProps = {
  value: "a value",
  handleChange: jest.fn(),
  placeholder: "a placeholder",
}
