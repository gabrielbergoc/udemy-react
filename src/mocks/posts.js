export const posts = [
  {
    id: 1,
    title: "post 1 title",
    body: "post 1 body",
    cover: "img/img1.png",
  },
  {
    id: 2,
    title: "post 2 title",
    body: "post 2 body",
    cover: "img/img2.png",
  },
  {
    id: 3,
    title: "post 3 title",
    body: "post 3 body",
    cover: "img/img3.png",
  },
];

export const post = posts[0];
