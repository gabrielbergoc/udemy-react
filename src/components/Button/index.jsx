import P from 'prop-types';
import './styles.css';

const Button = ({ text, type = 'submit', onClick, disabled = false }) => (
  <button type={type} onClick={onClick} className="button" disabled={disabled}>
    {text}
  </button>
);

Button.propTypes = {
  text: P.string.isRequired,
  type: P.string,
  onClick: P.func.isRequired,
  disabled: P.bool,
};

export default Button;
