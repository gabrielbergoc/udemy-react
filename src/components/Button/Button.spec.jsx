import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '.';
import { buttonProps } from '../../mocks/button';

describe('<Button />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button {...buttonProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with given text', () => {
    expect.assertions(1); // important in async tests
    render(<Button {...buttonProps} />);

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
  });

  it('should call prop function on click', () => {
    const fn = jest.fn();
    render(<Button {...buttonProps} onClick={fn} />);
    const button = screen.getByRole('button');

    userEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should be disabled if disabled=true', () => {
    render(<Button {...buttonProps} disabled={true} />);

    const button = screen.getByRole('button');

    expect(button).toBeDisabled();
  });

  it('should NOT be disabled if disabled=false', () => {
    render(<Button {...buttonProps} disabled={false} />);

    const button = screen.getByRole('button');

    expect(button).toBeEnabled();
  });
});
