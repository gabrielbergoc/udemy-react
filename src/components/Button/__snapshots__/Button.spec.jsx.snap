// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div>
  <button
    class="button"
    type="submit"
  >
    button text
  </button>
</div>
`;
